reframe/frontend/autodetect.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import json
+import os
+import tempfile
+
+import reframe as rfm
+import reframe.utility.osext as osext
+from reframe.core.logging import getlogger
+from reframe.core.runtime import runtime
+from reframe.core.schedulers import Job
+from reframe.utility.cpuinfo import cpuinfo
+
+
+def _load_info(filename):
+    try:
+        with open(filename) as fp:
+            return json.load(fp)
+    except OSError as e:
+        getlogger().warning(
+            f'could not load file: {filename!r}: {e}'
+        )
+        return {}
```

Comment:
Good catch, I'm not doing any validation for it now :-D

Reply:
The only problem that I see currently with this is that we most likely need to duplicate the part of the configuration schema that handles the processor info. Currently, this is part of the global configuration schema, but the files we are loading here have only the processor and device parts, so we need a separate schema file for those. I will try to see whether the "big" schema can include other subschemas from different files. I want to avoid the duplication in the schemas in all cases.

Reply:
Ok, I see that I can simply selects parts of the big schema to validate against. We're doing that already for validating the default values in the configuration.

Reply:
Ok, I've done it. We can simply pick a part of a loaded schema and validate something against it.

Reply:
I was a bit quick in responding. There is a side effect due to the way we have structured the schema. The solution is not that straightforward, because we are using cross-references in the schema. So, in practice, we need to create a custom schema on the fly with all the references that we need...

Reply:
The validation pays off. I already caught a small bug in how `numa_nodes` were reported for Macs ;-)